### ISSUE TITLE
Doesn't work anymore. Probably another checksum issue.

### ISSUE THREAD

**Issue by RoniJames** [ericshermancs/ZyBookComplete#2]
{"error":{"code":403,"message":"Invalid request"},"success":false}
is given for every question.

This has saved my life so many times already please if you can fix it I would love you forever.

**Comment by RoniJames** [comment]
I would also love it if you could give me some insight on how you're able to decode the checksum. I've been trying to see If I could figure it out but the obfuscated JavaScript is very hard to follow. Thank you!

**Comment by fm360** [comment]
I have the same issue

**Comment by washplant26** [comment]
I have the same issue, please fix, thank you!

**Comment by sooot** [comment]
I believe the issue lies here:

https://github.com/ericshermancs/ZyBookComplete/blob/03fe90e01e25a7418ed9a62e6f13854fca198fac/zybook.py#L12-L18

After looking at the obfuscated JS:

![screenshot](https://user-images.githubusercontent.com/14035185/130907984-a9fa2166-04db-46f9-93b0-27a2225d1510.png)

This is the line that creates the checksum, where:
- `r` is `content_resource/{activity_id}/activity`
- `T` is the timestamp in ISO format: `2021-08-26T01:23:45.678Z`
- `k` is the JWT token: `eyJhb...`
- **(new)** `O` is the activity_id as a string: `"123456"`
- **(new)** `M` is the "part" as an integer: `0`
- **(new)** `P` is the "complete" as a boolean: `false`
- `C` is the buildkey

**Comment by ptrtgr** [comment]
Thanks sooot! I couldn't figure out the obfuscated checksum logic. Alternatively, my EmberJS approach for autocompleting Zybooks currently works: https://github.com/ptrtgr/zybookfuck
